index.d.ts

```diff
@@ -0,0 +1,143 @@
+// Since react-native-udp acts as a RN compatible port of dgram, this type decl. has been copied from dgram
+declare module 'react-native-udp' {
+  import { AddressInfo } from 'net';
+  import * as dns from 'dns';
+  import * as events from 'events';
+
+  interface RemoteInfo {
+      address: string;
+      family: 'IPv4' | 'IPv6';
+      port: number;
+      size: number;
+  }
+
+  interface BindOptions {
+      port?: number;
+      address?: string;
+      exclusive?: boolean;
+      fd?: number;
+  }
+
+  type SocketType = 'udp4' | 'udp6';
+
+  interface SocketOptions {
+      type: SocketType;
+      reuseAddr?: boolean;
+      /**
+       * @default false
+       */
+      ipv6Only?: boolean;
+      recvBufferSize?: number;
+      sendBufferSize?: number;
+      lookup?: (hostname: string, options: dns.LookupOneOptions, callback: (err: NodeJS.ErrnoException | null, address: string, family: number) => void) => void;
+      debug?: boolean;
+  }
+
+  function createSocket(type: SocketType, callback?: (msg: Buffer, rinfo: RemoteInfo) => void): Socket;
+  function createSocket(options: SocketOptions, callback?: (msg: Buffer, rinfo: RemoteInfo) => void): Socket;
+
+  class Socket extends events.EventEmitter {
+      addMembership(multicastAddress: string, multicastInterface?: string): void;
+      address(): AddressInfo;
+      bind(port?: number, address?: string, callback?: (err: Error) => void): void;
+      bind(port?: number, callback?: (err: Error) => void): void;
+      bind(callback?: (err: Error) => void): void;
+      bind(options: BindOptions, callback?: (err: Error) => void): void;
+      close(callback?: () => void): void;
+      connect(port: number, address?: string, callback?: () => void): void;
+      connect(port: number, callback: () => void): void;
+      disconnect(): void;
+      dropMembership(multicastAddress: string, multicastInterface?: string): void;
+      getRecvBufferSize(): number;
+      getSendBufferSize(): number;
```

Comment:
These methods are not implemented yet, should we remove them from the types declaration? It might lead to confusion. What do you think?

Reply:
Yup, best to remove any dgram methods not mirrored in react-native-udp.
Will you edit this, if I do it might be later this week

Reply:
I am a bit busy right now with all this pandemic thing, some things are a bit more tedious to perform right now. I would comment all the unimplemented methods ASAP. If I finally find time and change something, I will update this "review".